lib/blueprinter/reflection.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Blueprinter
+  #
+  # Public methods for reflecting on a Blueprint.
+  #
+  module Reflection
+    #
+    # Returns a Hash of views keyed by name.
+    #
+    # Example:
+    #
+    #   widget_view = WidgetBlueprint.views[:default]
+    #   category = widget_view.associations[:category]
+    #   category.blueprint
+    #   => CategoryBlueprint
+    #   category.view
+    #   => :default
+    #
+    # @return [Hash<Symbol, Blueprinter::Reflection::View>]
+    #
+    def views
+      @views ||= view_collection.views.transform_values do |view|
+        View.new(self, view)
+      end
+    end
+
+    #
+    # Represents a view within a Blueprint.
+    #
+    class View
+      Field = Struct.new(:name, :display_name, :options)
+      Association = Struct.new(:name, :display_name, :blueprint, :view, :options)
+
+      def initialize(blueprint, view)
+        @blueprint = blueprint
+        @view = view
+      end
+
+      # @return [String] The view's name
+      def name
+        @view.name
+      end
+
+      #
+      # Returns a Hash of fields in this view (recursive) keyed by method name.
+      #
+      # @return [Hash<Symbol, Blueprinter::Reflection::View::Field>]
+      #
+      def fields
+        @fields ||= @view.fields.each_with_object(included(:fields)) do |(_name, field), obj|
+          unless field.options[:association] # rubocop:todo Style/IfUnlessModifier
+            obj[field.method] = Field.new(field.method, field.name, field.options)
+          end
+        end
+      end
+
+      #
+      # Returns a Hash of associations in this view (recursive) keyed by method name.
```

Comment:
Should we consider this recursive? When I read this initially, I had thought that we might be returning the full "tree" of associations (i.e. associations contained within top-level associations). 

Or are we just reiterating here due to the implementation of `#included`?

Reply:
Yeah, I think I left that comment from an original implementation that was actually recursive. Will change.